core/cache/machine.go

```diff
@@ -76,3 +70,61 @@ func (m *Machine) setDetails(details MachineChange) {
 		// TODO: publish config change...
 	}
 }
+
+// WatchApplicationLXDProfiles notifies if any of the following happen
+// relative to this machine:
+//     1. A new unit whose charm has an lxd profile is added.
+//     2. A unit being removed has a profile and other units
+//        exist on the machine.
+//     3. The lxdprofile of an application with a unit on this
+//        machine is added, removed, or exists.
+func (m *Machine) WatchApplicationLXDProfiles() (*MachineAppLXDProfileWatcher, error) {
+	units, err := m.Units()
+	if err != nil {
+		return nil, errors.Annotatef(err, "failed to get units to start MachineAppLXDProfileWatcher")
+	}
+	m.model.mu.Lock()
+	applications := make(map[string]appInfo)
+	for _, unit := range units {
+		_, found := applications[unit.details.Application]
```

Comment:
```suggestion
               appName := unit.details.Application
		_, found := applications[appName]
```

And use the variable alias to reduce noise and make it easier to read.

Reply:
done